pkg/controllers/machine_controller.go

```diff
@@ -285,6 +298,109 @@ func (r *MachineReconciler) startCheckAndEnqueueMachineState(ctx context.Context
 	}, r.resyncIntervalMachineState)
 }
 
+func (r *MachineReconciler) startGarbageCollector(ctx context.Context, log logr.Logger) {
+	wait.UntilWithContext(ctx, func(ctx context.Context) {
+		machines, err := r.machines.List(ctx)
+		if err != nil {
+			log.Error(err, "failed to list machines")
+			return
+		}
+
+		for _, machine := range machines {
+			if !slices.Contains(machine.Finalizers, MachineFinalizer) || machine.DeletedAt == nil {
+				continue
+			}
+
+			logger := log.WithValues("machineID", machine.ID)
+			if err := r.processMachineDeletion(ctx, logger, machine); err != nil {
+				logger.Error(err, "failed to garbage collect machine")
+			}
+		}
+
+	}, r.resyncIntervalGarbageCollector)
+}
+
+func (r *MachineReconciler) processMachineDeletion(ctx context.Context, log logr.Logger, machine *api.Machine) error {
+	isDeleting, err := r.deleteMachine(ctx, log, machine)
```

Comment:
Can we add a log here stating the the deletion is triggered?

Reply:
We have the option to include it. However, this log will be reiterated in the console following every resync interval, in conjunction with the "Triggering shutdown" log. We already possess logs that clearly illustrate whether a shutdown or domain destruction has taken place. Is this the anticipated behavior?